tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,168 @@
+            env_inserts += "-e CALICO_IPV4POOL_IPIP=%s " % ipip
+        prefix, _, suffix = base_command.partition("-e")
+        command = prefix + env_inserts + "-e" + suffix
+
+        # Start calico-docker
+        self.host.execute(command)
+
+        if not success_expected:
+            # check for "Calico node failed to start"
+            expected_string = "Calico node failed to start"
+            check = functools.partial(self.assert_calico_node_log_contains, expected_string)
+            retry_until_success(check, 5, ex_class=AssertionError)
+            return
+
+        # Check we started OK
+        expected_string = "Calico node started successfully"
+        check = functools.partial(self.assert_calico_node_log_contains, expected_string)
+        retry_until_success(check, 5, ex_class=AssertionError)
+        # check the expected pool is present
+        pools_output = self.host.calicoctl("get ippool -o yaml")
+        pools_dict = yaml.safe_load(pools_output)
+        cidrs = [pool['metadata']['cidr'] for pool in pools_dict]
+        # Convert to canonical form
+        value = str(netaddr.IPNetwork(value))
+        assert value in cidrs, "Didn't find %s in %s" % (value, cidrs)
+
+        # Dump pools and attempt to load them with calicoctl (to confirm consistency)
+        self.host.calicoctl("get ippool -o yaml > pool")
+        self.host.calicoctl("apply -f pool")
+
+        assert len(pools_dict) == exp_num_pools, \
+            "Expected %s pools, found %s. %s" % (exp_num_pools, len(pools_dict), pools_dict)
+
+        # Grab the pool of interest
+        pool = pools_dict[cidrs.index(value)]
+        other_pool = None
+        # And grab the other pool if any
+        if len(pools_dict) > 1:
+            pools_dict.remove(pool)
+            other_pool = pools_dict[0]
+        # Check IPIP setting if we're doing IPv4
+        if ipip in ["cross-subnet", "always"] and param == "CALICO_IPV4POOL_CIDR":
+            assert pool['spec']['ipip']['enabled'] is True, \
+                "Didn't find ipip enabled in pool %s" % pool
+            assert pool['spec']['ipip']['mode'] == ipip, \
+                "Didn't find ipip mode in pool %s" % pool
+        if ipip in [None, "off"] or param == "CALICO_IPV6POOL_CIDR":
+            assert 'ipip' not in pool['spec']
+        if ipip in ["cross-subnet", "always"] and param == "CALICO_IPV6POOL_CIDR":
+            assert other_pool['spec']['ipip']['enabled'] is True, \
+                "Didn't find ipip enabled in pool %s" % pool
+            assert other_pool['spec']['ipip']['mode'] == ipip, \
+                "Didn't find ipip mode in pool %s" % pool
+
+        # Check NAT setting
+        assert pool['spec']['nat-outgoing'] is True, "Didn't find nat enabled in pool %s" % pool
+
+    def test_no_default_pools(self):
+        """
+        Test that NO_DEFAULT_POOLS works correctly
+        """
+        # Start calico-docker
+        self.host.calicoctl("node run --no-default-ippools")
```

Comment:
docker_host has a start_calico_node() method that allows you to pass in extra options.  It also takes care of using the correct container name.